src/src/browser.cpp

```diff
@@ -63,7 +63,7 @@ BrowserPrivate::BrowserPrivate(Browser *browser, AbstractServer *server, const Q
 bool BrowserPrivate::updateService(const QByteArray &fqName)
 {
     // Split the FQDN into service name and type
-    int index = fqName.indexOf('.');
+    int index = fqName.indexOf('_') - 1;
```

Comment:
perhaps this should be "._" (assume empty service names shouldn't be supported)?